cookbooks/bcpc/recipes/cobbler.rb

```diff
@@ -20,11 +20,21 @@
 require 'digest/sha2'
 
 make_config('cobbler-web-user', "cobbler")
-make_config('cobbler-web-password', secure_password)
-make_config('cobbler-root-password', secure_password)
-make_config('cobbler-root-password-salted', "#{get_config('cobbler-root-password')}".crypt("$6$" + rand(36**8).to_s(36)) )
+
+create_databag("os")
+
+chef_vault_secret "cobbler" do
+  data_bag 'os'
+  raw_data({ 'web-password' => secure_password, 'root-password' => secure_password })
+  admins node[:hostname]
+  search '*:*'
+  action :nothing
+end.run_action(:create_if_missing)
+
+cobbler = get_secret("os","cobbler")
+make_config('cobbler-root-password-salted', "#{cobbler['root-password']}".crypt("$6$" + rand(36**8).to_s(36)) )
```

Comment:
If cobbler is nil, the script will fail at the next step : make_config(), and whole chef run should exit by this failure